daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -280,6 +280,15 @@ object Speedy {
         ctrl = CtrlExpr(sexpr),
       )
     }
+
+    // Construct a machine from an SExpr. This is useful when you don’t have
+    // an update expression and build’s behavior of applying the expression to
+    // a token is not appropriate.
+    def fromSExpr(
+        sexpr: SExpr,
+        checkSubmitterInMaintainers: Boolean,
```

Comment:
Does that flag make sense in this context?

Reply:
> The REPL for DAML-LF has to solve a very similar problem for evaluating `f x y ...`, which is possible, IIRC. Do we have duplicate code now?

Good point, I’ve refactored the other initialization methods to use `fromSExpr`.

Reply:
> Does that flag make sense in this context?

It does, there is nothing stopping you from having an update expression here. You are just not forced to use one.

Reply:
Makes sense.